ui/lib/core/addon/helpers/sync-destinations.ts

```diff
@@ -74,8 +80,12 @@ export function destinationTypes(): Array<DestinationType> {
   return SYNC_DESTINATIONS.map((d) => d.type);
 }
 
-export function findDestination(type: DestinationType | undefined): SyncDestination | undefined {
-  return SYNC_DESTINATIONS.find((d) => d.type === type);
+export function findDestination(type: DestinationType) {
+  const destination = SYNC_DESTINATIONS.find((d) => d.type === type);
+  if (!destination) {
+    throw new Error(`Destination not found for type: ${type}`);
+  }
+  return destination;
```

Comment:
It was getting repetitive doing `findDestination(type)?.icon` for example so I added in an error check if an unsupported type is passed in.